### PR TITLE
Fix misspell and add api/listener test

### PR DESCRIPTION
**Pull request by applemin**
fix misspell and add api/listener_test.go

### CODE REVIEW

File: api/listener.go [hunk 1]
```diff
@@ -48,6 +49,7 @@ func (cl *chainListener) Start() error {
 			case <-cl.cancelChan:
 				// notify all responders to exit
 				cl.streamMap.Range(func(key, _ interface{}) bool {
+					fmt.Print(key)
```
Comment (CoderZhi) [inline api/listener.go]: revert

File: api/listener_test.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+func TestServer_Start(t *testing.T) {
+	require := require.New(t)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	responder := mock_responder.NewMockResponder(ctrl)
+	responder.EXPECT().Respond(gomock.Any()).Return(nil).AnyTimes()
+	lister := NewChainListener()
+	lister.AddResponder(responder)
+	err := lister.Start()
```
Comment (CoderZhi) [inline api/listener_test.go]: `require.NoError(listener.Start())`

File: api/listener_test.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	"github.com/stretchr/testify/require"
+)
+
+func TestServer_Start(t *testing.T) {
+	require := require.New(t)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	responder := mock_responder.NewMockResponder(ctrl)
+	responder.EXPECT().Respond(gomock.Any()).Return(nil).AnyTimes()
+	lister := NewChainListener()
```
Comment (CoderZhi) [inline api/listener_test.go]: typo

File: api/listener_test.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	lister.AddResponder(responder)
+	err := lister.Start()
+	for _, i := range []int{1, 10, 100, 1000} {
+		block1 := makeBlock(t, i)
+		lister.HandleBlock(block1)
+	}
+	require.NoError(err)
+}
+
+func makeBlock(tb testing.TB, n int) *block.Block {
```
Comment (CoderZhi) [inline api/listener_test.go]: `func makeBlock(n int) (*block.Block, error) {`

File: api/listener_test.go [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+	"github.com/iotexproject/iotex-core/test/mock/mock_responder"
+	"github.com/stretchr/testify/require"
+)
+
+func TestServer_Start(t *testing.T) {
+	require := require.New(t)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	responder := mock_responder.NewMockResponder(ctrl)
+	responder.EXPECT().Respond(gomock.Any()).Return(nil).AnyTimes()
```
Comment (CoderZhi) [inline api/listener_test.go]: we are suppose to verify something in this mock function

File: e2etest/rewarding_test.go [hunk 6]
```diff
@@ -125,7 +125,7 @@ func TestBlockReward(t *testing.T) {
 
 func TestBlockEpochReward(t *testing.T) {
 	// TODO: fix the test
-	t.Skip()
+	//t.Skip()
```
Comment (CoderZhi) [inline e2etest/rewarding_test.go]: why?

Comment (applemin) [reply e2etest/rewarding_test.go]: incorrect operation

File: api/listener.go (hunk 1 above)
Comment (applemin) [reply api/listener.go]: OK

File: e2etest/rewarding_test.go (hunk 6 above)
Comment (yutongp) [reply e2etest/rewarding_test.go]: https://golang.org/pkg/testing/#hdr-Skipping
test skipping is supported in golang std. Please check you go environment. This should not be an incorrect operation.

Comment (applemin) [reply e2etest/rewarding_test.go]: It was opened when I tested the project. It was my personal misoperation

Comment (applemin) [reply e2etest/rewarding_test.go]: sorry

File: api/listener_test.go (hunk 3 above)
Comment (applemin) [reply api/listener_test.go]: Recovered

File: e2etest/rewarding_test.go (hunk 6 above)
Comment (applemin) [reply e2etest/rewarding_test.go]: Recovered

File: api/listener_test.go (hunk 2 above)
Comment (applemin) [reply api/listener_test.go]: Recovered

File: api/listener.go (hunk 1 above)
Comment (applemin) [reply api/listener.go]: Recovered

**Pull request closed without merging**